src/validator.js

```diff
@@ -256,49 +256,41 @@ export default class Validator {
 
         var ext = extname(this.packagePath);
         if (extensions.hasOwnProperty(ext)) {
-          return Promise.resolve(extensions[ext]);
+          return extensions[ext];
         }
 
         log.error('All attempts to lookup addon type failed');
         this.collector.addError(messages.TYPE_NOT_DETERMINED);
-        return Promise.resolve(null);
+        return null;
       });
   }
 
   checkFileExists(filepath, _lstat=lstat) {
     var invalidMessage = new Error(
       `Path "${filepath}" is not a file or does not exist.`);
-    return new Promise((resolve, reject) => {
-      return _lstat(filepath)
-        .then((stats) => {
-          if (stats.isFile() === true) {
-            resolve();
-          } else {
-            reject(invalidMessage);
-          }
-        })
-        .catch((err) => {
-          if (err.code !== 'ENOENT') {
-            reject(err);
-          } else {
-            reject(invalidMessage);
-          }
-        });
-    });
+    return _lstat(filepath)
+      .then((stats) => {
+        if (stats.isFile() === true) {
+          return;
```

Comment:
Again, now I just feel like it's harder to quickly grasp this returns a Promise. Given nearly everything we do does it's not too bad, but I feel like something is lost here. More code feels more clear in this case.